src/targets/gpu/deconvolution.cpp

```diff
@@ -55,14 +55,40 @@ argument miopen_deconvolution::compute(context& ctx,
                                        const shape& output_shape,
                                        const std::vector<argument>& args) const
 {
-    auto x_desc = make_tensor(reshape_if_1d(args[0].get_shape()));
-    auto w_desc = make_tensor(reshape_if_1d(args[1].get_shape()));
-    auto y_desc = make_tensor(reshape_if_1d(output_shape));
+    auto x_desc                = make_tensor(reshape_if_1d(args[0].get_shape()));
+    auto w_desc                = make_tensor(reshape_if_1d(args[1].get_shape()));
+    auto y_desc                = make_tensor(reshape_if_1d(output_shape));
+    auto* miopen_stream_handle = ctx.get_stream().get_miopen();
+    auto workspace_size        = args[2].get_shape().bytes();
 
+#ifdef MIGRAPHX_HAS_FIND_2_API
+    {
+        const miopenTensorArgument_t tensor_args[3] = {
+            {miopenTensorConvolutionX, nullptr, args[0].implicit()},
+            {miopenTensorConvolutionW, nullptr, args[1].implicit()},
+            {miopenTensorConvolutionY, nullptr, args[3].implicit()},
+        };
+
+        if(solution_ptr.get() == nullptr)
+            MIGRAPHX_THROW("MIOpen Deconvolution : Load MIOpen Solution before running it");
+
+        auto status = miopenRunSolution(miopen_stream_handle,
+                                        solution_ptr.get(),
+                                        3,
+                                        tensor_args,
+                                        args[2].implicit(),
+                                        workspace_size);
+        if(status != miopenStatusSuccess)
+            MIGRAPHX_THROW("MIOpen Deconvolution: running convolution using find_2.0 failed");
+
+        return args[3];
+    }
```

Comment:
These should all be combined into the same class. Similiar to how we combine rocblas gemms with `rcoblas_gemm`:

```cpp
template<class Op>
struct miopen_convolution
{
    Op op;
    bool int8_x4_format = false;
    shared<convolution_descriptor> cd;
    miopenConvFwdAlgorithm_t algo{};
#ifdef MIGRAPHX_HAS_FIND_2_API
    value::binary solution_object{};
    shared<miopen_solution> solution_ptr = nullptr;
#endif
    uint64_t solution_id = 0;

    template <class Self, class F>
    static auto reflect(Self& self, F f)
    {
        // TODO: Add algo
        return pack_join(migraphx::reflect(self.op, f),
                         pack(
#ifdef MIGRAPHX_HAS_FIND_2_API
                             f(self.solution_object, "solution_object"),
#endif
                             f(self.int8_x4_format, "int8_x4_format")));
    }
    ...
};
```

We could inline everything and then implement the [`compile`](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/src/include/migraphx/operation.hpp#L491) method to get the workspace, which means we can avoid including this header elsewhere(I think only lowering uses these headers). I also plan to implement this for NHWC.

Reply:
Unified all into one single base class. 

`compile` method, are you saying i should rename `find` method to `compile` ?